code/game/gamemodes/cult/cult_items.dm

```diff
@@ -72,6 +72,173 @@
+	else
+		to_chat(loc, "<span class='notice'>You lower the [src] and prepare to swing it normally.</span>")
+
+/obj/item/twohanded/required/cult_bastard/pickup(mob/living/user)
+	. = ..()
+	if(!iscultist(user))
+		if(!is_servant_of_ratvar(user))
+			to_chat(user, "<span class='cultlarge'>\"I wouldn't advise that.\"</span>")
+			to_chat(user, "<span class='warning'>An overwhelming sense of nausea overpowers you!</span>")
+			user.Dizzy(80)
+			user.Knockdown(30)
+			return
+		else
+			to_chat(user, "<span class='cultlarge'>\"One of Ratvar's toys is trying to play with things [user.p_they()] shouldn't. Cute.\"</span>")
+			to_chat(user, "<span class='userdanger'>A horrible force yanks at your arm!</span>")
+			user.emote("scream")
+			user.apply_damage(30, BRUTE, pick("l_arm", "r_arm"))
+			user.Knockdown(50)
+			return
+	linked_action = new(user)
+	linked_action.Grant(user, src)
+	user.update_icons()
+
+/obj/item/twohanded/required/cult_bastard/dropped(mob/user)
+	. = ..()
+	linked_action.Remove(user)
+	user.update_icons()
+
+/obj/item/twohanded/required/cult_bastard/IsReflect()
+	if(spinning)
+		playsound(src, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 75, 1)
+		return 1
+	else
+		..()
+
+/obj/item/twohanded/required/cult_bastard/hit_reaction(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", final_block_chance = 0, damage = 0, attack_type = MELEE_ATTACK)
+	if(prob(final_block_chance))
+		if(attack_type == PROJECTILE_ATTACK)
+			owner.visible_message("<span class='danger'>[owner] deflects [attack_text] with [src]!</span>")
+			playsound(src, pick('sound/weapons/bulletflyby.ogg', 'sound/weapons/bulletflyby2.ogg', 'sound/weapons/bulletflyby3.ogg'), 100, 1)
+			return 1
+		else
+			playsound(src, 'sound/weapons/parry.ogg', 75, 1)
+			owner.visible_message("<span class='danger'>[owner] parries [attack_text] with [src]!</span>")
+			return 1
+	return 0
+
+/obj/item/twohanded/required/cult_bastard/afterattack(atom/target, mob/user, proximity, click_parameters)
+	. = ..()
+	if(dash_toggled && charges)
+		dash(user, target)
+		return
+	if(!proximity)
+		return
+	if(ishuman(target))
+		var/mob/living/carbon/human/H = target
+		if(H.stat != CONSCIOUS)
+			var/obj/item/device/soulstone/SS = new /obj/item/device/soulstone(src)
+			SS.attack(H, user)
+			shards += SS
+		return
+	if(istype(target, /obj/structure/constructshell) && shards.len)
+		var/obj/item/device/soulstone/SS = shards[1]
+		if(SS.transfer_soul("CONSTRUCT",target,user))
+			shards -= SS
+			qdel(SS)
+
+/obj/item/twohanded/required/cult_bastard/proc/charge()
+	charges = Clamp(charges + 1, 0, 1)
+	playsound(src, 'sound/magic/exit_blood.ogg', 25, 1)
+	if(istype(loc, /mob/living))
+		to_chat(loc, "<span class='cultitalic'>[src] is ready for another blood jaunt.</span>")
+
+/obj/item/twohanded/required/cult_bastard/proc/dash(mob/user, atom/target)
```

Comment:
You need to figure out some solution other than straight copypasting dash weapons

Reply:
Well it was either that or making this a child of dashing and straight copypasting 2H code. 

I can clean it up a little but at the end of the day dash weapons use the dash proc, and this sword uses a dash.

Reply:
Make a dummy dash weapon object and put it inside the sword then call the dash from that when you click with the 2 handed one?

Reply:
From a lay person's perspective that sounds a lot more problematic than just copying the proc itself but if you insist, I guess the sword is already full of soul shards might as well stuff another sword inside it too.

Reply:
make the dash behaviour a datum or component, see the stuff cyberboss does